doc: kernel: revise doc for k_thread_stack_alloc and k_thread_create (#88200)
zephyrproject-rtos/zephyr: doc/kernel/services/threads/index.rst

  macros, the return value of :c:macro:`K_KERNEL_STACK_SIZEOF()` for that
  object.

Alternatively, the stack area may be dynamically allocated using

> :c:func:`k_thread_stack_alloc`. In that situation, the caller should remember
> to use :c:func:`k_thread_stack_free`, after the thread has been joined, to

Reviewer: ```suggestion
:c:func:`k_thread_stack_alloc`. In that situation, the caller must
use :c:func:`k_thread_stack_free`, after the thread has been joined, to
```
maybe a bit stronger wording?

Author: Fixed!
